{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3609ce09",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Import the libraries \n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "os.chdir('/Users/rebaviles21/QBIO_490_rebecca/analysis_data')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2bd50ec5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Your version of cptac (1.1.2) is out-of-date. Latest is 1.5.7. Please run 'pip install --upgrade cptac' to update it. (/Users/rebaviles21/miniconda3/lib/python3.11/threading.py, line 975)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          \r"
     ]
    }
   ],
   "source": [
    "import cptac\n",
    "cptac.download(dataset=\"Ccrcc\")\n",
    "ccrcc = cptac.Ccrcc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b5e1813c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Sample_Tumor_Normal</th>\n",
       "      <th>tumor/normal</th>\n",
       "      <th>gender</th>\n",
       "      <th>age</th>\n",
       "      <th>height_in_cm</th>\n",
       "      <th>height_in_inch</th>\n",
       "      <th>weight_in_kg</th>\n",
       "      <th>weight_in_lb</th>\n",
       "      <th>BMI</th>\n",
       "      <th>race</th>\n",
       "      <th>...</th>\n",
       "      <th>histologic_type_of_normal_tissue</th>\n",
       "      <th>slide_is_free_of_tumor</th>\n",
       "      <th>consistent_with_local_pathology_report</th>\n",
       "      <th>findings_not_consistent_with_local_pathology_report</th>\n",
       "      <th>weight_in_mg</th>\n",
       "      <th>minutes_clamp_1_to_collection</th>\n",
       "      <th>minutes_clamp_2_to_collection</th>\n",
       "      <th>minutes_collection_to_frozen</th>\n",
       "      <th>consistent_with_diagnostic_report</th>\n",
       "      <th>patient_medications</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00004</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>TN</td>\n",
       "      <td>Male</td>\n",
       "      <td>72</td>\n",
       "      <td>170.0</td>\n",
       "      <td>67.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>145.0</td>\n",
       "      <td>22.80</td>\n",
       "      <td>White</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>484.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rivaroxaban,tylenol,Aspirin,glycolax,colace,zocor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00010</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>TN</td>\n",
       "      <td>Male</td>\n",
       "      <td>30</td>\n",
       "      <td>177.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>236.0</td>\n",
       "      <td>34.15</td>\n",
       "      <td>White</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>575.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Rivaroxaban,Esomeprazole ,Tramadol</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00011</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>TN</td>\n",
       "      <td>Female</td>\n",
       "      <td>63</td>\n",
       "      <td>180.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>196.0</td>\n",
       "      <td>27.47</td>\n",
       "      <td>White</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>272.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Multi Vitamin,Levothyroxine Sodium,Ibandronate...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00026</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>TN</td>\n",
       "      <td>Female</td>\n",
       "      <td>65</td>\n",
       "      <td>163.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>28.23</td>\n",
       "      <td>White</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>212.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Levothyroxine Sodium,Cyproheptadine HCL,Citrac...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00079</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>TN</td>\n",
       "      <td>Male</td>\n",
       "      <td>49</td>\n",
       "      <td>175.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>116.0</td>\n",
       "      <td>256.0</td>\n",
       "      <td>37.88</td>\n",
       "      <td>White</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>675.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28</td>\n",
       "      <td>Yes</td>\n",
       "      <td>ibuprofen,Norco,miralax</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01646.N</th>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>renal cortex, medulla</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>256.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01648.N</th>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>renal cortex</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>288.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01649.N</th>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>renal medulla, pelvis</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>301.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01651.N</th>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>renal cortex, medulla</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>280.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01808.N</th>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>renal cortex</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>195.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>194 rows × 171 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        Sample_Tumor_Normal tumor/normal  gender  age  height_in_cm  \\\n",
       "Patient_ID                                                                \n",
       "C3L-00004                 Tumor           TN    Male   72         170.0   \n",
       "C3L-00010                 Tumor           TN    Male   30         177.0   \n",
       "C3L-00011                 Tumor           TN  Female   63         180.0   \n",
       "C3L-00026                 Tumor           TN  Female   65         163.0   \n",
       "C3L-00079                 Tumor           TN    Male   49         175.0   \n",
       "...                         ...          ...     ...  ...           ...   \n",
       "C3N-01646.N              Normal          NaN     NaN  NaN           NaN   \n",
       "C3N-01648.N              Normal          NaN     NaN  NaN           NaN   \n",
       "C3N-01649.N              Normal          NaN     NaN  NaN           NaN   \n",
       "C3N-01651.N              Normal          NaN     NaN  NaN           NaN   \n",
       "C3N-01808.N              Normal          NaN     NaN  NaN           NaN   \n",
       "\n",
       "Name         height_in_inch  weight_in_kg  weight_in_lb    BMI   race  ...  \\\n",
       "Patient_ID                                                             ...   \n",
       "C3L-00004              67.0          66.0         145.0  22.80  White  ...   \n",
       "C3L-00010              70.0         107.0         236.0  34.15  White  ...   \n",
       "C3L-00011              71.0          89.0         196.0  27.47  White  ...   \n",
       "C3L-00026              64.0          75.0         165.0  28.23  White  ...   \n",
       "C3L-00079              69.0         116.0         256.0  37.88  White  ...   \n",
       "...                     ...           ...           ...    ...    ...  ...   \n",
       "C3N-01646.N             NaN           NaN           NaN    NaN    NaN  ...   \n",
       "C3N-01648.N             NaN           NaN           NaN    NaN    NaN  ...   \n",
       "C3N-01649.N             NaN           NaN           NaN    NaN    NaN  ...   \n",
       "C3N-01651.N             NaN           NaN           NaN    NaN    NaN  ...   \n",
       "C3N-01808.N             NaN           NaN           NaN    NaN    NaN  ...   \n",
       "\n",
       "Name        histologic_type_of_normal_tissue slide_is_free_of_tumor  \\\n",
       "Patient_ID                                                            \n",
       "C3L-00004                                NaN                    NaN   \n",
       "C3L-00010                                NaN                    NaN   \n",
       "C3L-00011                                NaN                    NaN   \n",
       "C3L-00026                                NaN                    NaN   \n",
       "C3L-00079                                NaN                    NaN   \n",
       "...                                      ...                    ...   \n",
       "C3N-01646.N            renal cortex, medulla                    Yes   \n",
       "C3N-01648.N                     renal cortex                    Yes   \n",
       "C3N-01649.N            renal medulla, pelvis                    Yes   \n",
       "C3N-01651.N            renal cortex, medulla                    Yes   \n",
       "C3N-01808.N                     renal cortex                    Yes   \n",
       "\n",
       "Name        consistent_with_local_pathology_report  \\\n",
       "Patient_ID                                           \n",
       "C3L-00004                                      Yes   \n",
       "C3L-00010                                      Yes   \n",
       "C3L-00011                                      Yes   \n",
       "C3L-00026                                      Yes   \n",
       "C3L-00079                                      Yes   \n",
       "...                                            ...   \n",
       "C3N-01646.N                                    Yes   \n",
       "C3N-01648.N                                    Yes   \n",
       "C3N-01649.N                                    Yes   \n",
       "C3N-01651.N                                    Yes   \n",
       "C3N-01808.N                                    Yes   \n",
       "\n",
       "Name        findings_not_consistent_with_local_pathology_report  weight_in_mg  \\\n",
       "Patient_ID                                                                      \n",
       "C3L-00004                                                  NaN          484.0   \n",
       "C3L-00010                                                  NaN          575.0   \n",
       "C3L-00011                                                  NaN          272.0   \n",
       "C3L-00026                                                  NaN          212.0   \n",
       "C3L-00079                                                  NaN          675.0   \n",
       "...                                                        ...            ...   \n",
       "C3N-01646.N                                                NaN          256.0   \n",
       "C3N-01648.N                                                NaN          288.0   \n",
       "C3N-01649.N                                                NaN          301.0   \n",
       "C3N-01651.N                                                NaN          280.0   \n",
       "C3N-01808.N                                                NaN          195.0   \n",
       "\n",
       "Name        minutes_clamp_1_to_collection minutes_clamp_2_to_collection  \\\n",
       "Patient_ID                                                                \n",
       "C3L-00004                             NaN                           NaN   \n",
       "C3L-00010                             NaN                           NaN   \n",
       "C3L-00011                             NaN                           NaN   \n",
       "C3L-00026                            13.0                           NaN   \n",
       "C3L-00079                             NaN                           NaN   \n",
       "...                                   ...                           ...   \n",
       "C3N-01646.N                           NaN                           NaN   \n",
       "C3N-01648.N                           NaN                           NaN   \n",
       "C3N-01649.N                           NaN                           NaN   \n",
       "C3N-01651.N                           NaN                           NaN   \n",
       "C3N-01808.N                           NaN                           NaN   \n",
       "\n",
       "Name        minutes_collection_to_frozen consistent_with_diagnostic_report  \\\n",
       "Patient_ID                                                                   \n",
       "C3L-00004                             21                               Yes   \n",
       "C3L-00010                             17                               Yes   \n",
       "C3L-00011                             18                               Yes   \n",
       "C3L-00026                             13                               Yes   \n",
       "C3L-00079                             28                               Yes   \n",
       "...                                  ...                               ...   \n",
       "C3N-01646.N                           20                               NaN   \n",
       "C3N-01648.N                            8                               NaN   \n",
       "C3N-01649.N                           29                               NaN   \n",
       "C3N-01651.N                           22                               NaN   \n",
       "C3N-01808.N                           29                               NaN   \n",
       "\n",
       "Name                                       patient_medications  \n",
       "Patient_ID                                                      \n",
       "C3L-00004    rivaroxaban,tylenol,Aspirin,glycolax,colace,zocor  \n",
       "C3L-00010                   Rivaroxaban,Esomeprazole ,Tramadol  \n",
       "C3L-00011    Multi Vitamin,Levothyroxine Sodium,Ibandronate...  \n",
       "C3L-00026    Levothyroxine Sodium,Cyproheptadine HCL,Citrac...  \n",
       "C3L-00079                              ibuprofen,Norco,miralax  \n",
       "...                                                        ...  \n",
       "C3N-01646.N                                                NaN  \n",
       "C3N-01648.N                                                NaN  \n",
       "C3N-01649.N                                                NaN  \n",
       "C3N-01651.N                                                NaN  \n",
       "C3N-01808.N                                                NaN  \n",
       "\n",
       "[194 rows x 171 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_data = ccrcc.get_clinical()\n",
    "clinical_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0fac84cf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAC</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00004</th>\n",
       "      <td>-0.304302</td>\n",
       "      <td>0.641447</td>\n",
       "      <td>-0.000025</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.207831</td>\n",
       "      <td>-0.364128</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.203886</td>\n",
       "      <td>-0.217934</td>\n",
       "      <td>0.216894</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.064343</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.094441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.021827</td>\n",
       "      <td>0.133927</td>\n",
       "      <td>0.237280</td>\n",
       "      <td>0.114409</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00010</th>\n",
       "      <td>1.195915</td>\n",
       "      <td>0.194620</td>\n",
       "      <td>1.360294</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.126956</td>\n",
       "      <td>-0.572843</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.596546</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.221696</td>\n",
       "      <td>...</td>\n",
       "      <td>0.112064</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.072262</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.205642</td>\n",
       "      <td>0.182434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.201374</td>\n",
       "      <td>-0.068340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00011</th>\n",
       "      <td>-0.286155</td>\n",
       "      <td>-0.780455</td>\n",
       "      <td>-0.101089</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.292629</td>\n",
       "      <td>0.035812</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.300863</td>\n",
       "      <td>...</td>\n",
       "      <td>0.136957</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.279732</td>\n",
       "      <td>0.695116</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.316298</td>\n",
       "      <td>-0.009772</td>\n",
       "      <td>-0.019653</td>\n",
       "      <td>-0.095339</td>\n",
       "      <td>0.008961</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00026</th>\n",
       "      <td>0.135730</td>\n",
       "      <td>0.404286</td>\n",
       "      <td>0.261384</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.155568</td>\n",
       "      <td>0.336311</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.709046</td>\n",
       "      <td>0.244198</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.013139</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.157541</td>\n",
       "      <td>0.526188</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.120501</td>\n",
       "      <td>0.054559</td>\n",
       "      <td>-0.313236</td>\n",
       "      <td>0.062194</td>\n",
       "      <td>0.052825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00079</th>\n",
       "      <td>-0.123959</td>\n",
       "      <td>-0.677773</td>\n",
       "      <td>-0.362547</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.187605</td>\n",
       "      <td>-0.320026</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.300148</td>\n",
       "      <td>-0.153216</td>\n",
       "      <td>0.229676</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.058953</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.152341</td>\n",
       "      <td>0.072886</td>\n",
       "      <td>0.068182</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.178869</td>\n",
       "      <td>0.266290</td>\n",
       "      <td>-0.028647</td>\n",
       "      <td>0.003682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01646.N</th>\n",
       "      <td>-0.533915</td>\n",
       "      <td>0.352304</td>\n",
       "      <td>-0.297937</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.049507</td>\n",
       "      <td>0.421582</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.108650</td>\n",
       "      <td>...</td>\n",
       "      <td>0.148773</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.215333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.042826</td>\n",
       "      <td>-0.071776</td>\n",
       "      <td>-0.189634</td>\n",
       "      <td>-0.078515</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01648.N</th>\n",
       "      <td>-0.732322</td>\n",
       "      <td>0.111213</td>\n",
       "      <td>-0.877605</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.058466</td>\n",
       "      <td>-0.241223</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.166769</td>\n",
       "      <td>...</td>\n",
       "      <td>0.390944</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.059535</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.366396</td>\n",
       "      <td>-0.070845</td>\n",
       "      <td>0.248763</td>\n",
       "      <td>0.140119</td>\n",
       "      <td>0.200438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01649.N</th>\n",
       "      <td>0.318404</td>\n",
       "      <td>0.065235</td>\n",
       "      <td>-0.261260</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.013386</td>\n",
       "      <td>0.648428</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.009946</td>\n",
       "      <td>...</td>\n",
       "      <td>0.198995</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.074344</td>\n",
       "      <td>-0.633999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.020812</td>\n",
       "      <td>0.162851</td>\n",
       "      <td>-0.571237</td>\n",
       "      <td>0.203220</td>\n",
       "      <td>0.111064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01651.N</th>\n",
       "      <td>-0.531707</td>\n",
       "      <td>0.525514</td>\n",
       "      <td>-1.065982</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.022212</td>\n",
       "      <td>-0.064736</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.335925</td>\n",
       "      <td>-0.017308</td>\n",
       "      <td>...</td>\n",
       "      <td>0.388512</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.237719</td>\n",
       "      <td>-0.181989</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.008758</td>\n",
       "      <td>0.198887</td>\n",
       "      <td>0.088718</td>\n",
       "      <td>0.185857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01808.N</th>\n",
       "      <td>0.089660</td>\n",
       "      <td>0.706365</td>\n",
       "      <td>-0.192663</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.033791</td>\n",
       "      <td>0.291573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.019725</td>\n",
       "      <td>...</td>\n",
       "      <td>0.154932</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.195492</td>\n",
       "      <td>-0.602745</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.150427</td>\n",
       "      <td>0.208886</td>\n",
       "      <td>-0.197532</td>\n",
       "      <td>0.145788</td>\n",
       "      <td>-0.052599</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>194 rows × 11710 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name             A1BG      A1CF       A2M  A4GALT      AAAS      AACS  AADAC  \\\n",
       "Patient_ID                                                                     \n",
       "C3L-00004   -0.304302  0.641447 -0.000025     NaN  0.207831 -0.364128    NaN   \n",
       "C3L-00010    1.195915  0.194620  1.360294     NaN  0.126956 -0.572843    NaN   \n",
       "C3L-00011   -0.286155 -0.780455 -0.101089     NaN  0.292629  0.035812    NaN   \n",
       "C3L-00026    0.135730  0.404286  0.261384     NaN  0.155568  0.336311    NaN   \n",
       "C3L-00079   -0.123959 -0.677773 -0.362547     NaN  0.187605 -0.320026    NaN   \n",
       "...               ...       ...       ...     ...       ...       ...    ...   \n",
       "C3N-01646.N -0.533915  0.352304 -0.297937     NaN  0.049507  0.421582    NaN   \n",
       "C3N-01648.N -0.732322  0.111213 -0.877605     NaN  0.058466 -0.241223    NaN   \n",
       "C3N-01649.N  0.318404  0.065235 -0.261260     NaN  0.013386  0.648428    NaN   \n",
       "C3N-01651.N -0.531707  0.525514 -1.065982     NaN  0.022212 -0.064736    NaN   \n",
       "C3N-01808.N  0.089660  0.706365 -0.192663     NaN  0.033791  0.291573    NaN   \n",
       "\n",
       "Name            AADAT     AAED1     AAGAB  ...    ZSWIM8  ZSWIM9      ZW10  \\\n",
       "Patient_ID                                 ...                               \n",
       "C3L-00004   -1.203886 -0.217934  0.216894  ... -0.064343     NaN -0.094441   \n",
       "C3L-00010   -1.596546       NaN  0.221696  ...  0.112064     NaN  0.072262   \n",
       "C3L-00011         NaN       NaN  0.300863  ...  0.136957     NaN  0.279732   \n",
       "C3L-00026         NaN  0.709046  0.244198  ... -0.013139     NaN  0.157541   \n",
       "C3L-00079   -1.300148 -0.153216  0.229676  ... -0.058953     NaN  0.152341   \n",
       "...               ...       ...       ...  ...       ...     ...       ...   \n",
       "C3N-01646.N       NaN       NaN  0.108650  ...  0.148773     NaN  0.215333   \n",
       "C3N-01648.N       NaN       NaN  0.166769  ...  0.390944     NaN  0.059535   \n",
       "C3N-01649.N       NaN       NaN -0.009946  ...  0.198995     NaN  0.074344   \n",
       "C3N-01651.N       NaN -0.335925 -0.017308  ...  0.388512     NaN  0.237719   \n",
       "C3N-01808.N       NaN       NaN  0.019725  ...  0.154932     NaN  0.195492   \n",
       "\n",
       "Name           ZWILCH     ZWINT      ZXDC    ZYG11B       ZYX     ZZEF1  \\\n",
       "Patient_ID                                                                \n",
       "C3L-00004         NaN       NaN -0.021827  0.133927  0.237280  0.114409   \n",
       "C3L-00010         NaN       NaN -0.205642  0.182434       NaN  0.201374   \n",
       "C3L-00011    0.695116       NaN  0.316298 -0.009772 -0.019653 -0.095339   \n",
       "C3L-00026    0.526188       NaN -0.120501  0.054559 -0.313236  0.062194   \n",
       "C3L-00079    0.072886  0.068182       NaN  0.178869  0.266290 -0.028647   \n",
       "...               ...       ...       ...       ...       ...       ...   \n",
       "C3N-01646.N       NaN       NaN  0.042826 -0.071776 -0.189634 -0.078515   \n",
       "C3N-01648.N       NaN       NaN  0.366396 -0.070845  0.248763  0.140119   \n",
       "C3N-01649.N -0.633999       NaN -0.020812  0.162851 -0.571237  0.203220   \n",
       "C3N-01651.N -0.181989       NaN       NaN -0.008758  0.198887  0.088718   \n",
       "C3N-01808.N -0.602745       NaN -0.150427  0.208886 -0.197532  0.145788   \n",
       "\n",
       "Name             ZZZ3  \n",
       "Patient_ID             \n",
       "C3L-00004         NaN  \n",
       "C3L-00010   -0.068340  \n",
       "C3L-00011    0.008961  \n",
       "C3L-00026    0.052825  \n",
       "C3L-00079    0.003682  \n",
       "...               ...  \n",
       "C3N-01646.N       NaN  \n",
       "C3N-01648.N  0.200438  \n",
       "C3N-01649.N  0.111064  \n",
       "C3N-01651.N  0.185857  \n",
       "C3N-01808.N -0.052599  \n",
       "\n",
       "[194 rows x 11710 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_data = ccrcc.get_proteomics()\n",
    "protein_data.columns = protein_data.columns.get_level_values(0) \n",
    "protein_data\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "3d4ae11d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A3GALT2</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>A4GNT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAC</th>\n",
       "      <th>...</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDA</th>\n",
       "      <th>ZXDB</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11A</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00004</th>\n",
       "      <td>0.995336</td>\n",
       "      <td>16.677828</td>\n",
       "      <td>353.263362</td>\n",
       "      <td>0.046634</td>\n",
       "      <td>0.031027</td>\n",
       "      <td>17.196885</td>\n",
       "      <td>0.896109</td>\n",
       "      <td>15.831130</td>\n",
       "      <td>2.938550</td>\n",
       "      <td>0.019029</td>\n",
       "      <td>...</td>\n",
       "      <td>2.622184</td>\n",
       "      <td>5.415676</td>\n",
       "      <td>5.293887</td>\n",
       "      <td>3.731920</td>\n",
       "      <td>12.541358</td>\n",
       "      <td>2.677696</td>\n",
       "      <td>15.595194</td>\n",
       "      <td>11.401808</td>\n",
       "      <td>9.976986</td>\n",
       "      <td>23.334614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00010</th>\n",
       "      <td>0.679400</td>\n",
       "      <td>16.682712</td>\n",
       "      <td>359.078446</td>\n",
       "      <td>0.077350</td>\n",
       "      <td>0.068617</td>\n",
       "      <td>13.560508</td>\n",
       "      <td>1.743989</td>\n",
       "      <td>16.690257</td>\n",
       "      <td>3.154143</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>2.873604</td>\n",
       "      <td>9.209695</td>\n",
       "      <td>3.669353</td>\n",
       "      <td>2.560578</td>\n",
       "      <td>13.570779</td>\n",
       "      <td>4.097483</td>\n",
       "      <td>15.449647</td>\n",
       "      <td>11.550727</td>\n",
       "      <td>9.432121</td>\n",
       "      <td>25.724814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00011</th>\n",
       "      <td>0.354549</td>\n",
       "      <td>0.245606</td>\n",
       "      <td>222.075350</td>\n",
       "      <td>0.060736</td>\n",
       "      <td>0.273536</td>\n",
       "      <td>1.321499</td>\n",
       "      <td>0.172369</td>\n",
       "      <td>18.757568</td>\n",
       "      <td>6.942752</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>7.998655</td>\n",
       "      <td>28.780560</td>\n",
       "      <td>2.801800</td>\n",
       "      <td>2.503315</td>\n",
       "      <td>10.209840</td>\n",
       "      <td>0.178842</td>\n",
       "      <td>11.670596</td>\n",
       "      <td>11.342045</td>\n",
       "      <td>6.763858</td>\n",
       "      <td>32.090615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00026</th>\n",
       "      <td>2.543775</td>\n",
       "      <td>16.347532</td>\n",
       "      <td>228.282343</td>\n",
       "      <td>0.085684</td>\n",
       "      <td>0.152020</td>\n",
       "      <td>7.868391</td>\n",
       "      <td>1.448911</td>\n",
       "      <td>17.648610</td>\n",
       "      <td>6.175010</td>\n",
       "      <td>0.031078</td>\n",
       "      <td>...</td>\n",
       "      <td>2.754936</td>\n",
       "      <td>12.639323</td>\n",
       "      <td>5.262024</td>\n",
       "      <td>2.796869</td>\n",
       "      <td>10.718552</td>\n",
       "      <td>0.800663</td>\n",
       "      <td>15.887414</td>\n",
       "      <td>11.788588</td>\n",
       "      <td>8.169953</td>\n",
       "      <td>24.752283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00079</th>\n",
       "      <td>4.355205</td>\n",
       "      <td>4.858958</td>\n",
       "      <td>275.090167</td>\n",
       "      <td>0.106359</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.863003</td>\n",
       "      <td>2.338081</td>\n",
       "      <td>15.480282</td>\n",
       "      <td>4.584445</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>7.497914</td>\n",
       "      <td>14.400917</td>\n",
       "      <td>2.907591</td>\n",
       "      <td>2.417113</td>\n",
       "      <td>10.127549</td>\n",
       "      <td>3.442177</td>\n",
       "      <td>12.807428</td>\n",
       "      <td>17.494840</td>\n",
       "      <td>9.733803</td>\n",
       "      <td>24.528238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01646.N</th>\n",
       "      <td>1.287655</td>\n",
       "      <td>11.076241</td>\n",
       "      <td>228.037113</td>\n",
       "      <td>0.106505</td>\n",
       "      <td>0.066692</td>\n",
       "      <td>8.195971</td>\n",
       "      <td>0.982361</td>\n",
       "      <td>18.086400</td>\n",
       "      <td>10.225658</td>\n",
       "      <td>0.040903</td>\n",
       "      <td>...</td>\n",
       "      <td>2.338694</td>\n",
       "      <td>3.274051</td>\n",
       "      <td>3.500041</td>\n",
       "      <td>3.223773</td>\n",
       "      <td>15.663483</td>\n",
       "      <td>3.219438</td>\n",
       "      <td>17.530849</td>\n",
       "      <td>6.471017</td>\n",
       "      <td>11.860328</td>\n",
       "      <td>25.489666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01648.N</th>\n",
       "      <td>1.435986</td>\n",
       "      <td>9.772280</td>\n",
       "      <td>291.393930</td>\n",
       "      <td>0.172457</td>\n",
       "      <td>0.214180</td>\n",
       "      <td>8.955806</td>\n",
       "      <td>1.838108</td>\n",
       "      <td>14.597847</td>\n",
       "      <td>6.598100</td>\n",
       "      <td>0.056297</td>\n",
       "      <td>...</td>\n",
       "      <td>1.983803</td>\n",
       "      <td>1.486905</td>\n",
       "      <td>3.865995</td>\n",
       "      <td>3.197000</td>\n",
       "      <td>15.581449</td>\n",
       "      <td>5.774722</td>\n",
       "      <td>17.131855</td>\n",
       "      <td>10.133624</td>\n",
       "      <td>15.256983</td>\n",
       "      <td>22.669748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01649.N</th>\n",
       "      <td>1.082318</td>\n",
       "      <td>8.378616</td>\n",
       "      <td>249.779349</td>\n",
       "      <td>0.100673</td>\n",
       "      <td>0.160751</td>\n",
       "      <td>7.419323</td>\n",
       "      <td>2.290458</td>\n",
       "      <td>17.666594</td>\n",
       "      <td>11.108717</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>2.007450</td>\n",
       "      <td>2.590684</td>\n",
       "      <td>3.805016</td>\n",
       "      <td>2.943559</td>\n",
       "      <td>17.274857</td>\n",
       "      <td>5.856484</td>\n",
       "      <td>16.392044</td>\n",
       "      <td>6.775269</td>\n",
       "      <td>12.881048</td>\n",
       "      <td>25.672959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01651.N</th>\n",
       "      <td>0.770924</td>\n",
       "      <td>15.539566</td>\n",
       "      <td>273.743429</td>\n",
       "      <td>0.208978</td>\n",
       "      <td>0.185383</td>\n",
       "      <td>12.206999</td>\n",
       "      <td>2.377283</td>\n",
       "      <td>13.946563</td>\n",
       "      <td>6.920178</td>\n",
       "      <td>0.022739</td>\n",
       "      <td>...</td>\n",
       "      <td>1.861395</td>\n",
       "      <td>2.904921</td>\n",
       "      <td>3.089317</td>\n",
       "      <td>2.529058</td>\n",
       "      <td>14.839389</td>\n",
       "      <td>5.801733</td>\n",
       "      <td>18.372506</td>\n",
       "      <td>9.149521</td>\n",
       "      <td>13.918938</td>\n",
       "      <td>24.578885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01808.N</th>\n",
       "      <td>0.622108</td>\n",
       "      <td>16.125722</td>\n",
       "      <td>172.780700</td>\n",
       "      <td>0.091174</td>\n",
       "      <td>0.025541</td>\n",
       "      <td>9.702769</td>\n",
       "      <td>2.006484</td>\n",
       "      <td>17.649919</td>\n",
       "      <td>10.172484</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.713981</td>\n",
       "      <td>3.014346</td>\n",
       "      <td>3.411386</td>\n",
       "      <td>3.001192</td>\n",
       "      <td>18.717445</td>\n",
       "      <td>12.680195</td>\n",
       "      <td>19.327095</td>\n",
       "      <td>6.351638</td>\n",
       "      <td>14.957318</td>\n",
       "      <td>24.678258</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>185 rows × 19275 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name             A1BG       A1CF         A2M     A2ML1   A3GALT2     A4GALT  \\\n",
       "Patient_ID                                                                    \n",
       "C3L-00004    0.995336  16.677828  353.263362  0.046634  0.031027  17.196885   \n",
       "C3L-00010    0.679400  16.682712  359.078446  0.077350  0.068617  13.560508   \n",
       "C3L-00011    0.354549   0.245606  222.075350  0.060736  0.273536   1.321499   \n",
       "C3L-00026    2.543775  16.347532  228.282343  0.085684  0.152020   7.868391   \n",
       "C3L-00079    4.355205   4.858958  275.090167  0.106359  0.000000   6.863003   \n",
       "...               ...        ...         ...       ...       ...        ...   \n",
       "C3N-01646.N  1.287655  11.076241  228.037113  0.106505  0.066692   8.195971   \n",
       "C3N-01648.N  1.435986   9.772280  291.393930  0.172457  0.214180   8.955806   \n",
       "C3N-01649.N  1.082318   8.378616  249.779349  0.100673  0.160751   7.419323   \n",
       "C3N-01651.N  0.770924  15.539566  273.743429  0.208978  0.185383  12.206999   \n",
       "C3N-01808.N  0.622108  16.125722  172.780700  0.091174  0.025541   9.702769   \n",
       "\n",
       "Name            A4GNT       AAAS       AACS     AADAC  ...    ZWILCH  \\\n",
       "Patient_ID                                             ...             \n",
       "C3L-00004    0.896109  15.831130   2.938550  0.019029  ...  2.622184   \n",
       "C3L-00010    1.743989  16.690257   3.154143  0.000000  ...  2.873604   \n",
       "C3L-00011    0.172369  18.757568   6.942752  0.000000  ...  7.998655   \n",
       "C3L-00026    1.448911  17.648610   6.175010  0.031078  ...  2.754936   \n",
       "C3L-00079    2.338081  15.480282   4.584445  0.000000  ...  7.497914   \n",
       "...               ...        ...        ...       ...  ...       ...   \n",
       "C3N-01646.N  0.982361  18.086400  10.225658  0.040903  ...  2.338694   \n",
       "C3N-01648.N  1.838108  14.597847   6.598100  0.056297  ...  1.983803   \n",
       "C3N-01649.N  2.290458  17.666594  11.108717  0.000000  ...  2.007450   \n",
       "C3N-01651.N  2.377283  13.946563   6.920178  0.022739  ...  1.861395   \n",
       "C3N-01808.N  2.006484  17.649919  10.172484  0.000000  ...  1.713981   \n",
       "\n",
       "Name             ZWINT      ZXDA      ZXDB       ZXDC     ZYG11A     ZYG11B  \\\n",
       "Patient_ID                                                                    \n",
       "C3L-00004     5.415676  5.293887  3.731920  12.541358   2.677696  15.595194   \n",
       "C3L-00010     9.209695  3.669353  2.560578  13.570779   4.097483  15.449647   \n",
       "C3L-00011    28.780560  2.801800  2.503315  10.209840   0.178842  11.670596   \n",
       "C3L-00026    12.639323  5.262024  2.796869  10.718552   0.800663  15.887414   \n",
       "C3L-00079    14.400917  2.907591  2.417113  10.127549   3.442177  12.807428   \n",
       "...                ...       ...       ...        ...        ...        ...   \n",
       "C3N-01646.N   3.274051  3.500041  3.223773  15.663483   3.219438  17.530849   \n",
       "C3N-01648.N   1.486905  3.865995  3.197000  15.581449   5.774722  17.131855   \n",
       "C3N-01649.N   2.590684  3.805016  2.943559  17.274857   5.856484  16.392044   \n",
       "C3N-01651.N   2.904921  3.089317  2.529058  14.839389   5.801733  18.372506   \n",
       "C3N-01808.N   3.014346  3.411386  3.001192  18.717445  12.680195  19.327095   \n",
       "\n",
       "Name               ZYX      ZZEF1       ZZZ3  \n",
       "Patient_ID                                    \n",
       "C3L-00004    11.401808   9.976986  23.334614  \n",
       "C3L-00010    11.550727   9.432121  25.724814  \n",
       "C3L-00011    11.342045   6.763858  32.090615  \n",
       "C3L-00026    11.788588   8.169953  24.752283  \n",
       "C3L-00079    17.494840   9.733803  24.528238  \n",
       "...                ...        ...        ...  \n",
       "C3N-01646.N   6.471017  11.860328  25.489666  \n",
       "C3N-01648.N  10.133624  15.256983  22.669748  \n",
       "C3N-01649.N   6.775269  12.881048  25.672959  \n",
       "C3N-01651.N   9.149521  13.918938  24.578885  \n",
       "C3N-01808.N   6.351638  14.957318  24.678258  \n",
       "\n",
       "[185 rows x 19275 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna_data = ccrcc.get_transcriptomics()\n",
    "rna_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4b824875",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAC</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>...</th>\n",
       "      <th>histologic_type_of_normal_tissue</th>\n",
       "      <th>slide_is_free_of_tumor</th>\n",
       "      <th>consistent_with_local_pathology_report</th>\n",
       "      <th>findings_not_consistent_with_local_pathology_report</th>\n",
       "      <th>weight_in_mg</th>\n",
       "      <th>minutes_clamp_1_to_collection</th>\n",
       "      <th>minutes_clamp_2_to_collection</th>\n",
       "      <th>minutes_collection_to_frozen</th>\n",
       "      <th>consistent_with_diagnostic_report</th>\n",
       "      <th>patient_medications</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00004</th>\n",
       "      <td>-0.304302</td>\n",
       "      <td>0.641447</td>\n",
       "      <td>-0.000025</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.207831</td>\n",
       "      <td>-0.364128</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.203886</td>\n",
       "      <td>-0.217934</td>\n",
       "      <td>0.216894</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>484.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rivaroxaban,tylenol,Aspirin,glycolax,colace,zocor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00010</th>\n",
       "      <td>1.195915</td>\n",
       "      <td>0.194620</td>\n",
       "      <td>1.360294</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.126956</td>\n",
       "      <td>-0.572843</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.596546</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.221696</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>575.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Rivaroxaban,Esomeprazole ,Tramadol</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00011</th>\n",
       "      <td>-0.286155</td>\n",
       "      <td>-0.780455</td>\n",
       "      <td>-0.101089</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.292629</td>\n",
       "      <td>0.035812</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.300863</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>272.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Multi Vitamin,Levothyroxine Sodium,Ibandronate...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00026</th>\n",
       "      <td>0.135730</td>\n",
       "      <td>0.404286</td>\n",
       "      <td>0.261384</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.155568</td>\n",
       "      <td>0.336311</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.709046</td>\n",
       "      <td>0.244198</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>212.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Levothyroxine Sodium,Cyproheptadine HCL,Citrac...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00079</th>\n",
       "      <td>-0.123959</td>\n",
       "      <td>-0.677773</td>\n",
       "      <td>-0.362547</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.187605</td>\n",
       "      <td>-0.320026</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.300148</td>\n",
       "      <td>-0.153216</td>\n",
       "      <td>0.229676</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>675.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28</td>\n",
       "      <td>Yes</td>\n",
       "      <td>ibuprofen,Norco,miralax</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01646.N</th>\n",
       "      <td>-0.533915</td>\n",
       "      <td>0.352304</td>\n",
       "      <td>-0.297937</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.049507</td>\n",
       "      <td>0.421582</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.108650</td>\n",
       "      <td>...</td>\n",
       "      <td>renal cortex, medulla</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>256.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01648.N</th>\n",
       "      <td>-0.732322</td>\n",
       "      <td>0.111213</td>\n",
       "      <td>-0.877605</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.058466</td>\n",
       "      <td>-0.241223</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.166769</td>\n",
       "      <td>...</td>\n",
       "      <td>renal cortex</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>288.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01649.N</th>\n",
       "      <td>0.318404</td>\n",
       "      <td>0.065235</td>\n",
       "      <td>-0.261260</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.013386</td>\n",
       "      <td>0.648428</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.009946</td>\n",
       "      <td>...</td>\n",
       "      <td>renal medulla, pelvis</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>301.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01651.N</th>\n",
       "      <td>-0.531707</td>\n",
       "      <td>0.525514</td>\n",
       "      <td>-1.065982</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.022212</td>\n",
       "      <td>-0.064736</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.335925</td>\n",
       "      <td>-0.017308</td>\n",
       "      <td>...</td>\n",
       "      <td>renal cortex, medulla</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>280.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01808.N</th>\n",
       "      <td>0.089660</td>\n",
       "      <td>0.706365</td>\n",
       "      <td>-0.192663</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.033791</td>\n",
       "      <td>0.291573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.019725</td>\n",
       "      <td>...</td>\n",
       "      <td>renal cortex</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>195.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>194 rows × 11881 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name             A1BG      A1CF       A2M  A4GALT      AAAS      AACS  AADAC  \\\n",
       "Patient_ID                                                                     \n",
       "C3L-00004   -0.304302  0.641447 -0.000025     NaN  0.207831 -0.364128    NaN   \n",
       "C3L-00010    1.195915  0.194620  1.360294     NaN  0.126956 -0.572843    NaN   \n",
       "C3L-00011   -0.286155 -0.780455 -0.101089     NaN  0.292629  0.035812    NaN   \n",
       "C3L-00026    0.135730  0.404286  0.261384     NaN  0.155568  0.336311    NaN   \n",
       "C3L-00079   -0.123959 -0.677773 -0.362547     NaN  0.187605 -0.320026    NaN   \n",
       "...               ...       ...       ...     ...       ...       ...    ...   \n",
       "C3N-01646.N -0.533915  0.352304 -0.297937     NaN  0.049507  0.421582    NaN   \n",
       "C3N-01648.N -0.732322  0.111213 -0.877605     NaN  0.058466 -0.241223    NaN   \n",
       "C3N-01649.N  0.318404  0.065235 -0.261260     NaN  0.013386  0.648428    NaN   \n",
       "C3N-01651.N -0.531707  0.525514 -1.065982     NaN  0.022212 -0.064736    NaN   \n",
       "C3N-01808.N  0.089660  0.706365 -0.192663     NaN  0.033791  0.291573    NaN   \n",
       "\n",
       "Name            AADAT     AAED1     AAGAB  ...  \\\n",
       "Patient_ID                                 ...   \n",
       "C3L-00004   -1.203886 -0.217934  0.216894  ...   \n",
       "C3L-00010   -1.596546       NaN  0.221696  ...   \n",
       "C3L-00011         NaN       NaN  0.300863  ...   \n",
       "C3L-00026         NaN  0.709046  0.244198  ...   \n",
       "C3L-00079   -1.300148 -0.153216  0.229676  ...   \n",
       "...               ...       ...       ...  ...   \n",
       "C3N-01646.N       NaN       NaN  0.108650  ...   \n",
       "C3N-01648.N       NaN       NaN  0.166769  ...   \n",
       "C3N-01649.N       NaN       NaN -0.009946  ...   \n",
       "C3N-01651.N       NaN -0.335925 -0.017308  ...   \n",
       "C3N-01808.N       NaN       NaN  0.019725  ...   \n",
       "\n",
       "Name         histologic_type_of_normal_tissue  slide_is_free_of_tumor  \\\n",
       "Patient_ID                                                              \n",
       "C3L-00004                                 NaN                     NaN   \n",
       "C3L-00010                                 NaN                     NaN   \n",
       "C3L-00011                                 NaN                     NaN   \n",
       "C3L-00026                                 NaN                     NaN   \n",
       "C3L-00079                                 NaN                     NaN   \n",
       "...                                       ...                     ...   \n",
       "C3N-01646.N             renal cortex, medulla                     Yes   \n",
       "C3N-01648.N                      renal cortex                     Yes   \n",
       "C3N-01649.N             renal medulla, pelvis                     Yes   \n",
       "C3N-01651.N             renal cortex, medulla                     Yes   \n",
       "C3N-01808.N                      renal cortex                     Yes   \n",
       "\n",
       "Name         consistent_with_local_pathology_report  \\\n",
       "Patient_ID                                            \n",
       "C3L-00004                                       Yes   \n",
       "C3L-00010                                       Yes   \n",
       "C3L-00011                                       Yes   \n",
       "C3L-00026                                       Yes   \n",
       "C3L-00079                                       Yes   \n",
       "...                                             ...   \n",
       "C3N-01646.N                                     Yes   \n",
       "C3N-01648.N                                     Yes   \n",
       "C3N-01649.N                                     Yes   \n",
       "C3N-01651.N                                     Yes   \n",
       "C3N-01808.N                                     Yes   \n",
       "\n",
       "Name         findings_not_consistent_with_local_pathology_report  \\\n",
       "Patient_ID                                                         \n",
       "C3L-00004                                                  NaN     \n",
       "C3L-00010                                                  NaN     \n",
       "C3L-00011                                                  NaN     \n",
       "C3L-00026                                                  NaN     \n",
       "C3L-00079                                                  NaN     \n",
       "...                                                        ...     \n",
       "C3N-01646.N                                                NaN     \n",
       "C3N-01648.N                                                NaN     \n",
       "C3N-01649.N                                                NaN     \n",
       "C3N-01651.N                                                NaN     \n",
       "C3N-01808.N                                                NaN     \n",
       "\n",
       "Name         weight_in_mg  minutes_clamp_1_to_collection  \\\n",
       "Patient_ID                                                 \n",
       "C3L-00004           484.0                            NaN   \n",
       "C3L-00010           575.0                            NaN   \n",
       "C3L-00011           272.0                            NaN   \n",
       "C3L-00026           212.0                           13.0   \n",
       "C3L-00079           675.0                            NaN   \n",
       "...                   ...                            ...   \n",
       "C3N-01646.N         256.0                            NaN   \n",
       "C3N-01648.N         288.0                            NaN   \n",
       "C3N-01649.N         301.0                            NaN   \n",
       "C3N-01651.N         280.0                            NaN   \n",
       "C3N-01808.N         195.0                            NaN   \n",
       "\n",
       "Name         minutes_clamp_2_to_collection  minutes_collection_to_frozen  \\\n",
       "Patient_ID                                                                 \n",
       "C3L-00004                              NaN                            21   \n",
       "C3L-00010                              NaN                            17   \n",
       "C3L-00011                              NaN                            18   \n",
       "C3L-00026                              NaN                            13   \n",
       "C3L-00079                              NaN                            28   \n",
       "...                                    ...                           ...   \n",
       "C3N-01646.N                            NaN                            20   \n",
       "C3N-01648.N                            NaN                             8   \n",
       "C3N-01649.N                            NaN                            29   \n",
       "C3N-01651.N                            NaN                            22   \n",
       "C3N-01808.N                            NaN                            29   \n",
       "\n",
       "Name         consistent_with_diagnostic_report  \\\n",
       "Patient_ID                                       \n",
       "C3L-00004                                  Yes   \n",
       "C3L-00010                                  Yes   \n",
       "C3L-00011                                  Yes   \n",
       "C3L-00026                                  Yes   \n",
       "C3L-00079                                  Yes   \n",
       "...                                        ...   \n",
       "C3N-01646.N                                NaN   \n",
       "C3N-01648.N                                NaN   \n",
       "C3N-01649.N                                NaN   \n",
       "C3N-01651.N                                NaN   \n",
       "C3N-01808.N                                NaN   \n",
       "\n",
       "Name                                       patient_medications  \n",
       "Patient_ID                                                      \n",
       "C3L-00004    rivaroxaban,tylenol,Aspirin,glycolax,colace,zocor  \n",
       "C3L-00010                   Rivaroxaban,Esomeprazole ,Tramadol  \n",
       "C3L-00011    Multi Vitamin,Levothyroxine Sodium,Ibandronate...  \n",
       "C3L-00026    Levothyroxine Sodium,Cyproheptadine HCL,Citrac...  \n",
       "C3L-00079                              ibuprofen,Norco,miralax  \n",
       "...                                                        ...  \n",
       "C3N-01646.N                                                NaN  \n",
       "C3N-01648.N                                                NaN  \n",
       "C3N-01649.N                                                NaN  \n",
       "C3N-01651.N                                                NaN  \n",
       "C3N-01808.N                                                NaN  \n",
       "\n",
       "[194 rows x 11881 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unite_data = pd.merge(protein_data, clinical_data, on='Patient_ID')\n",
    "unite_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "ad6ae647",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Name\n",
      "LDB3      1.911659\n",
      "BTBD7     1.814474\n",
      "GDF6      1.646855\n",
      "COX4I2    1.501131\n",
      "SNCB      1.494579\n",
      "dtype: float64\n"
     ]
    }
   ],
   "source": [
    "#Preprocessing and cleaning the data\n",
    "clinical_data = clinical_data[clinical_data['tumor_stage_pathological'].notna()]\n",
    "ccrcc_index = clinical_data.index\n",
    "protein_data = protein_data.loc[ccrcc_index]\n",
    "\n",
    "#clean up the stages CRCC\n",
    "stage1_mask = clinical_data['tumor_stage_pathological'] == \"Stage I\"\n",
    "\n",
    "stage3_mask = clinical_data['tumor_stage_pathological'] == \"Stage III\"\n",
    "\n",
    "#clean up for proteins \n",
    "protein_stage1 = protein_data.loc[stage1_mask]\n",
    "\n",
    "protein_stage3 = protein_data.loc[stage3_mask]\n",
    "\n",
    "\n",
    "#collect the mean values to prepare the between cleaned data\n",
    "first_protein = protein_stage1.mean()\n",
    "\n",
    "third_protein = protein_stage3.mean()\n",
    "\n",
    "\n",
    "#find the corresponding and most relevant data \n",
    "\n",
    "differential_expression = np.abs(first_protein - third_protein)\n",
    "\n",
    "#column created to find the proteins\n",
    "protein_sorting = differential_expression.sort_values(ascending = False) \n",
    "cleaned_protein_sorting = protein_sorting.dropna()\n",
    "\n",
    "#collecting using RNA and clinical \n",
    "five_proteins = cleaned_protein_sorting.head(5)\n",
    "\n",
    "print(five_proteins)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "1ea526a9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Name\n",
      "LRRC43      inf\n",
      "SYNPO2L     inf\n",
      "C1orf145    inf\n",
      "MAST1       inf\n",
      "SYTL5       inf\n",
      "dtype: float64\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/pandas/core/internals/blocks.py:366: RuntimeWarning: divide by zero encountered in log2\n",
      "  result = func(self.values, **kwargs)\n"
     ]
    }
   ],
   "source": [
    "# 2) Create a new dataframe of your selected features, where the rows are the patients\n",
    "#and the columns are the expression values of genes you selected in step 1 (X data).\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "rna_data = rna_data.loc[ccrcc_index]\n",
    "log_rna = np.log2(rna_data)\n",
    "\n",
    "#set up cleaned masks to find the appropriate genes\n",
    "\n",
    "genes_stage1 = log_rna.loc[stage1_mask]\n",
    "genes_stage3 = log_rna.loc[stage3_mask]\n",
    "\n",
    "genes_first = genes_stage1.mean()\n",
    "genes_third = genes_stage3.mean()\n",
    "#differential expression to find highest mean value of genes \n",
    "\n",
    "gene_differential_expression = np.abs(genes_first - genes_third)\n",
    "\n",
    "gene_sorting = gene_differential_expression.sort_values(ascending=False)\n",
    "cleaned_gene_sorting = gene_sorting.dropna()\n",
    "\n",
    "five_genes = cleaned_gene_sorting.head(5)\n",
    "\n",
    "print(five_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "efdecbdd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Name          LRRC43   SYNPO2L  C1orf145     MAST1     SYTL5\n",
      "Patient_ID                                                  \n",
      "C3L-00004  -3.675676 -5.311829 -1.807802 -4.312687 -6.237873\n",
      "C3L-00010  -0.978065 -4.944366 -1.077769 -6.167616 -2.922878\n",
      "C3L-00011  -4.731917 -4.823750 -0.542115 -3.725072 -3.557149\n",
      "C3L-00026   0.675903 -7.604106 -2.292724 -6.604964 -1.575954\n",
      "C3L-00079  -3.130040 -7.444265  0.059762 -1.490926 -5.048381\n",
      "...              ...       ...       ...       ...       ...\n",
      "C3N-01646  -3.382436 -3.919053 -2.678061 -4.504874 -1.906499\n",
      "C3N-01648  -3.127310 -3.634180 -1.352545 -3.535502 -6.367579\n",
      "C3N-01649  -2.907623 -4.084344 -2.039749 -3.457171 -1.855110\n",
      "C3N-01651  -2.990224 -4.474374 -1.970347 -2.890269 -4.230493\n",
      "C3N-01808  -2.446471 -5.220127 -0.809210 -2.220985 -4.561209\n",
      "\n",
      "[110 rows x 5 columns]\n"
     ]
    }
   ],
   "source": [
    "selected_genes = ['LRRC43', 'SYNPO2L', 'C1orf145', 'MAST1', 'SYTL5']\n",
    "chosen_selected = log_rna[selected_genes]\n",
    "print(chosen_selected)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "87cd81c7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Patient_ID\n",
      "C3L-00004    Stage III\n",
      "C3L-00010      Stage I\n",
      "C3L-00011     Stage IV\n",
      "C3L-00026      Stage I\n",
      "C3L-00079    Stage III\n",
      "               ...    \n",
      "C3N-01646    Stage III\n",
      "C3N-01648     Stage II\n",
      "C3N-01649    Stage III\n",
      "C3N-01651     Stage II\n",
      "C3N-01808      Stage I\n",
      "Name: tumor_stage_pathological, Length: 104, dtype: object\n"
     ]
    }
   ],
   "source": [
    "#3) create a separate list of the patients' cancer stage , (tumor_stage_pathological)\n",
    "cancer_stage = clinical_data['tumor_stage_pathological']\n",
    "\n",
    "#inconsistencies in step 6 with mistmatched due to not clean data\n",
    "cancer_mask = ['C3L-00097', 'C3L-00448', 'C3L-00796', 'C3L-01607', 'C3N-00313', 'C3N-01178']\n",
    "new_mask = ~cancer_stage.index.isin(cancer_mask)\n",
    "newCancer_stage= cancer_stage[new_mask]\n",
    "\n",
    "print(newCancer_stage)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "5a1450a0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Name          LRRC43   SYNPO2L  C1orf145     MAST1     SYTL5\n",
      "Patient_ID                                                  \n",
      "C3L-00004  -3.675676 -5.311829 -1.807802 -4.312687 -6.237873\n",
      "C3L-00010  -0.978065 -4.944366 -1.077769 -6.167616 -2.922878\n",
      "C3L-00011  -4.731917 -4.823750 -0.542115 -3.725072 -3.557149\n",
      "C3L-00026   0.675903 -7.604106 -2.292724 -6.604964 -1.575954\n",
      "C3L-00079  -3.130040 -7.444265  0.059762 -1.490926 -5.048381\n",
      "...              ...       ...       ...       ...       ...\n",
      "C3N-01646  -3.382436 -3.919053 -2.678061 -4.504874 -1.906499\n",
      "C3N-01648  -3.127310 -3.634180 -1.352545 -3.535502 -6.367579\n",
      "C3N-01649  -2.907623 -4.084344 -2.039749 -3.457171 -1.855110\n",
      "C3N-01651  -2.990224 -4.474374 -1.970347 -2.890269 -4.230493\n",
      "C3N-01808  -2.446471 -5.220127 -0.809210 -2.220985 -4.561209\n",
      "\n",
      "[104 rows x 5 columns]\n",
      "Patient_ID\n",
      "C3L-00004    Stage III\n",
      "C3L-00010      Stage I\n",
      "C3L-00011     Stage IV\n",
      "C3L-00026      Stage I\n",
      "C3L-00079    Stage III\n",
      "               ...    \n",
      "C3N-01646    Stage III\n",
      "C3N-01648     Stage II\n",
      "C3N-01649    Stage III\n",
      "C3N-01651     Stage II\n",
      "C3N-01808      Stage I\n",
      "Name: tumor_stage_pathological, Length: 104, dtype: object\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/73/qy1sl3md2fqf9_mcn_kfng800000gn/T/ipykernel_17689/2598799393.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  chosen_selected.replace([np.inf, -np.inf], np.nan, inplace = True)\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:767: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if not hasattr(array, \"sparse\") and array.dtypes.apply(is_sparse).any():\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:767: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if not hasattr(array, \"sparse\") and array.dtypes.apply(is_sparse).any():\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n"
     ]
    }
   ],
   "source": [
    "#4) scale and encode your features and target\n",
    "#import graphing libraries and tools \n",
    "\n",
    "from sklearn.preprocessing import StandardScaler, LabelEncoder\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.impute import SimpleImputer\n",
    "\n",
    "chosen_selected.replace([np.inf, -np.inf], np.nan, inplace = True)\n",
    "chosen_selected = chosen_selected.dropna()\n",
    "\n",
    "#scaling \n",
    "scaler = StandardScaler()\n",
    "scaled_feature = scaler. fit_transform(chosen_selected)\n",
    "\n",
    "#encoding \n",
    "label = LabelEncoder()\n",
    "label_encoder_features = label.fit_transform(cancer_stage)\n",
    "\n",
    "print(chosen_selected)\n",
    "print(newCancer_stage)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "10a074a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "#5) create a train split of the X and Y data with train_size = 0.7\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(chosen_selected, newCancer_stage, train_size=0.7, random_state=42)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "33c3c0f3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.375\n",
      "0.25\n",
      "0.40625\n",
      "0.5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:767: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if not hasattr(array, \"sparse\") and array.dtypes.apply(is_sparse).any():\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:767: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if not hasattr(array, \"sparse\") and array.dtypes.apply(is_sparse).any():\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but KNeighborsClassifier was fitted with feature names\n",
      "  warnings.warn(\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:767: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if not hasattr(array, \"sparse\") and array.dtypes.apply(is_sparse).any():\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:767: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if not hasattr(array, \"sparse\") and array.dtypes.apply(is_sparse).any():\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but DecisionTreeClassifier was fitted with feature names\n",
      "  warnings.warn(\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:767: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if not hasattr(array, \"sparse\") and array.dtypes.apply(is_sparse).any():\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:686: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.\n",
      "  warnings.warn(\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:767: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if not hasattr(array, \"sparse\") and array.dtypes.apply(is_sparse).any():\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but MLPClassifier was fitted with feature names\n",
      "  warnings.warn(\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:767: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if not hasattr(array, \"sparse\") and array.dtypes.apply(is_sparse).any():\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:767: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if not hasattr(array, \"sparse\") and array.dtypes.apply(is_sparse).any():\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but GaussianNB was fitted with feature names\n",
      "  warnings.warn(\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:605: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype):\n",
      "/Users/rebaviles21/miniconda3/lib/python3.11/site-packages/sklearn/utils/validation.py:614: FutureWarning: is_sparse is deprecated and will be removed in a future version. Check `isinstance(dtype, pd.SparseDtype)` instead.\n",
      "  if is_sparse(pd_dtype) or not is_extension_array_dtype(pd_dtype):\n"
     ]
    }
   ],
   "source": [
    "#6) Test the accuracy of the four models and find the average accuracy between them all\n",
    "\n",
    "#import the data needed for the models\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "from sklearn.tree import DecisionTreeClassifier\n",
    "from sklearn.neural_network import MLPClassifier\n",
    "from sklearn.naive_bayes import GaussianNB\n",
    "from sklearn.metrics import accuracy_score\n",
    "\n",
    "kneighbors_class = KNeighborsClassifier()\n",
    "decisiontree_class = DecisionTreeClassifier()\n",
    "mlp_class = MLPClassifier()\n",
    "gaussianNB_class = GaussianNB()\n",
    "\n",
    "amt_of_runs = 10\n",
    "accuracy = []\n",
    "totalScore = 0\n",
    "\n",
    "for _ in range(amt_of_runs):\n",
    "    X_train, X_test, y_train, y_test = train_test_split(chosen_selected, newCancer_stage, train_size=0.7, random_state=42)\n",
    "\n",
    "    \n",
    "    \n",
    "#KNeighbors Model \n",
    "\n",
    "kneighbors_class.fit(X_train, y_train)\n",
    "knn_predictor = kneighbors_class.predict(X_test)\n",
    "knn_accurate = accuracy_score(y_test, knn_predictor)\n",
    "knNeighbors = kneighbors_class.score(np.array(X_test), y_test)\n",
    "\n",
    "\n",
    "#DecisionTree Model\n",
    "\n",
    "decisiontree_class.fit(X_train, y_train)\n",
    "decisiontree_predictor = decisiontree_class.predict(X_test)\n",
    "decisiontree_accurate = accuracy_score(y_test, decisiontree_predictor)\n",
    "decisionTree = decisiontree_class.score(np.array(X_test), y_test)\n",
    "\n",
    "#MLPClassifier Model\n",
    "mlp_class.fit(X_train, y_train)\n",
    "mlp_predictor = mlp_class.predict(X_test)\n",
    "mlp_accurate = accuracy_score(y_test, mlp_predictor)\n",
    "mlp = mlp_class.score(np.array(X_test), y_test)\n",
    "\n",
    "#GaussianNB Model\n",
    "\n",
    "gaussianNB_class.fit(X_train, y_train)\n",
    "gaussianNB_predictor = gaussianNB_class.predict(X_test)\n",
    "gaussianNB_accurate = accuracy_score(y_test, gaussianNB_predictor)\n",
    "gaussianNB = gaussianNB_class.score(np.array(X_test), y_test)\n",
    "\n",
    "#set up the runs and store the accuracy scores\n",
    "print(knNeighbors)\n",
    "print(decisionTree)\n",
    "print(mlp)\n",
    "print(gaussianNB)\n",
    "\n",
    "#In this case, the Gaussian NB classifier is the best one! This is then followed by MLP classifier, KNNeigbors Classifier, and then the least acccurate was the Decision Tree\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "16228300",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "052460c7",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
